### PR TITLE
FIX: Adding form template to category dropdown can some times be empty

### PR DESCRIPTION
**Pull request by tgxworld**
Why this change?

A system test which was testing our ability to add a form template to a
category was flaky. This turned out to be a client side bug where the
`FormTemplateChooser` dropdown may not display any dropdown options if
the ajax request to fetch the categories form template has not been
completed when the dropdown is opened.

What does this change do?

Make use of select-kit's `triggerSearch` function to fetch the records
which will properly rerender the dropdown options.

### CODE REVIEW

**Comment by tgxworld** [comment]
Flaky test was detected in https://github.com/discourse/discourse/actions/runs/7280240675/job/19838257862

**Comment by nattsw** [comment]
Hmm @tgxworld you are seeing the test failure?

**Comment by tgxworld** [comment]
@nattsw Hmm didn't fail for me locally but I will have a look.

**Comment by tgxworld** [comment]
@nattsw Should be working now :)

File: app/assets/javascripts/select-kit/addon/components/form-template-chooser.js [hunk 1]
```diff
@@ -11,43 +11,57 @@ export default MultiSelectComponent.extend({
...
-        this.set("templates", sortedTemplates);
-        this.set("selectKit.options.disabled", true);
-      }
+  async _fetchTemplates() {
+    if (this.get("loadingTemplates")) {
+      return;
+    }
+
+    this.set("templatesLoaded", false);
+    this.set("loadingTemplates", true);
```
Comment (nattsw) [inline app/assets/javascripts/select-kit/addon/components/form-template-chooser.js:42]: Not a problem with this PR per se, but I guess it's a bit weird we have two flags to signify the same thing.

**Review by nattsw — APPROVED** [review]

**Pull request merged**